broker/broker.go

```diff
@@ -160,8 +160,10 @@ func (b *Broker) Deprovision(context context.Context, instanceID string, details
 	}
 
 	deployment, err := findDeployment(b.Compose, instanceName)
-	if err != nil {
+	if err == errDeploymentNotFound {
 		return spec, brokerapi.ErrInstanceDoesNotExist
+	} else if err != nil {
```

Comment:
Minor - do we need this else here? I think It can be like that which IMHO is more readable:
```
if err == errDeploymentNotFound {
  return spec, brokerapi.ErrInstanceDoesNotExist
}
if err != nil {
  return spec, err
}
```